src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -413,13 +450,57 @@ public SolutionState RemoveProjects(ArrayBuilder<ProjectId> projectIds)
         foreach (var projectId in projectIds)
             newDependencyGraph = newDependencyGraph.WithProjectRemoved(projectId);
 
+        var languageCountDeltas = new TemporaryArray<(string language, int count)>();
+        foreach (var projectId in projectIds)
+        {
+            AddLanguageCount(ref languageCountDeltas, _projectIdToProjectStateMap[projectId].Language, amount: -1);
+        }
+
         return this.Branch(
             solutionAttributes: newSolutionAttributes,
             projectIds: newProjectIds,
             idToProjectStateMap: newStateMap,
+            projectCountByLanguage: AddLanguageCounts(ProjectCountByLanguage, languageCountDeltas),
             dependencyGraph: newDependencyGraph);
     }
 
+    private static void AddLanguageCount(ref TemporaryArray<(string language, int count)> languageCounts, string language, int amount)
+    {
+        var index = languageCounts.IndexOf(static (c, language) => c.language == language, language);
+        if (index < 0)
+        {
+            languageCounts.Add((language, amount));
```

Comment:
note.... i would like an assert here.  Because if you call AddLangCount(... -1) and the lang is not in this array, we end up with a lang count of -1 in this array.  That clearly can only happen in a bogus case, but we should have a contrcat call finding that.

Reply:
Actually, its not a bogus case. That's a normal case when we are removing projects. Maybe languageCounts should be called languageCountDeltas

Reply:
Renamed for clarity.